packages/appium/lib/cli/utils.js

```diff
@@ -52,9 +52,24 @@ async function spinWith (json, msg, fn) {
   }
 }
 
+let createRingBuffer = function (length) {
+
+  let pointer = 0, buffer = [];
+
+  return {
+    getBuf () {return buffer;},
+    get (key) {return buffer[key];},
+    push (item) {
+      buffer[pointer] = item;
+      pointer = (length + pointer + 1) % length;
```

Comment:
I think it might be confusing if we start recording recent logs to the beginning of the buffer. I would rather shift the buffer to the left and delete the very first record in order to add the last one, so that the original order is preserved